### PR TITLE
[16.0][MIG] fieldservice_route

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by luke-stdev001** [comment]
I don't think this should be closed.

Quick question, are there any plans to work on features for route-optimisation/planning (eg. Portatour type solution - https://www.portatour.com/) with pgRouting, Graphhopper, etc.?

**Comment by max3903** [comment]
> I don't think this should be closed.
> 
> Quick question, are there any plans to work on features for route-optimisation/planning (eg. Portatour type solution - https://www.portatour.com/) with pgRouting, Graphhopper, etc.?

Yes, but the plan needs funding to be executed... As of now we do not have any customers to finance this feature. I only know one who did integrate with https://simpliroute.com

**Comment by luke-stdev001** [comment]
> > I don't think this should be closed.
> > Quick question, are there any plans to work on features for route-optimisation/planning (eg. Portatour type solution - https://www.portatour.com/) with pgRouting, Graphhopper, Tile38, etc.?
> 
> Yes, but the plan needs funding to be executed... As of now we do not have any customers to finance this feature. I only know one who did integrate with https://simpliroute.com

Hi @max3903 ,

Please email me at:

it@combuilt.com.au

to discuss funding.

**Review by stefan-tecnativa — APPROVED** [review]
Functionally tested, it looks good to me 👍️

File: fieldservice_route/models/fsm_order.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        if self.dayroute_id and not self.dayroute_id.order_ids:
+            self.dayroute_id.unlink()
+        return vals
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            location = self.env["fsm.location"].browse(vals.get("location_id"))
+            if not vals.get("fsm_route_id"):
+                vals.update({"fsm_route_id": location.fsm_route_id.id})
```
Comment (pedrobaeza) [inline fieldservice_route/models/fsm_order.py]:
```suggestion
            if not vals.get("fsm_route_id") and vals.get("location_id"):
                location = self.env["fsm.location"].browse(vals["location_id"])
                vals.update({"fsm_route_id": location.fsm_route_id.id})
```

File: fieldservice_route/models/fsm_route_dayroute.py [hunk 2]
```diff
@@ -13,7 +13,12 @@ class FSMRouteDayRoute(models.Model):
     _description = "Field Service Route Dayroute"
 
     name = fields.Char(required=True, copy=False, default=lambda self: _("New"))
-    person_id = fields.Many2one(comodel_name="fsm.person", string="Person")
+    person_id = fields.Many2one(
+        comodel_name="fsm.person",
+        string="Person",
+        compute="_compute_person_id",
+        store=True,
```
Comment (pedrobaeza) [inline fieldservice_route/models/fsm_route_dayroute.py:20]: This should be `readonly=False` for having a computed writable field and editable like before.

File: fieldservice_route/models/fsm_route_dayroute.py [hunk 3]
```diff
@@ -34,7 +39,9 @@ class FSMRouteDayRoute(models.Model):
     last_location_id = fields.Many2one(
         comodel_name="fsm.location", string="Last Location"
     )
-    date_start_planned = fields.Datetime(string="Planned Start Time")
+    date_start_planned = fields.Datetime(
+        string="Planned Start Time", compute="_compute_date_start_planned", store=True
```
Comment (pedrobaeza) [inline fieldservice_route/models/fsm_route_dayroute.py]: This should be `readonly=False` for having a computed writable field and editable like before.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
@pedrobaeza  I implemented your changes and rebased the branch with the latest 16.0 code.

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**